tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,283 @@
+        "etcd": {
+            "extraArgs":
+                {"heartbeat-interval": 500,
+                 "election-timeout": 5000},
+            "snapshot": False,
+            "backupConfig":
+                {"intervalHours": 12, "retention": 6, "type": "backupConfig"},
+            "creation": "12h",
+            "retention": "72h",
+            "type": "etcdService"},
+        "kubeApi": {
+            "alwaysPullImages": False,
+            "podSecurityPolicy": False,
+            "serviceNodePortRange": "30000-32767",
+            "type": "kubeAPIService"}},
+    "sshAgentAuth": False}
+
+@if_test_rbac
+def test_rbac_node_template_create(remove_resource):
+    client = get_user_client()
+    node_template = node_template_do(client)
+    templates = client.list_node_template(name=node_template.name)
+    assert len(templates) == 1
+    time.sleep(.5)
+    remove_resource(node_template)
+
+
+@if_test_rbac
+def test_rbac_node_template_list(remove_resource):
+    client = get_user_client()
+    node_template = node_template_do(client)
+    templates = client.list_node_template(name=node_template.name)
+    assert len(templates) == 1
+    # Admin should be able to list template
+    admin_client = get_admin_client()
+    templates = admin_client.list_node_template(name=node_template.name)
+    assert len(templates) == 1
+    # User 2 should not be able to list templates
+    user1, user1_token = create_user(get_admin_client())
+    user1_client = get_client_for_token(user1_token)
+    templates2 = user1_client.list_node_template(name=node_template.name)
+    assert len(templates2) == 0
+    time.sleep(.5)
+    remove_resource(node_template)
+
+
+@if_test_rbac
+def test_rbac_node_template_delete(remove_resource):
+    client = get_user_client()
+    node_template = node_template_do(client)
+    # User1 should be able to delete own template
+    client.delete(node_template)
+    templates = client.list_node_template(name=node_template.name)
+    assert len(templates) == 0
+    # Admin should be able to delete template created by user1
+    admin_client = get_admin_client()
+    node_template2 = node_template_do(client)
+    admin_client.delete(node_template2)
+    templates = client.list_node_template(name=node_template2.name)
+    assert len(templates) == 0
+    # User 2 should not be able to delete template created by user1
+    user1, user1_token = create_user(get_admin_client())
+    user1_client = get_client_for_token(user1_token)
+    node_template3 = node_template_do(client)
+    with pytest.raises(ApiError) as e:
+        user1_client.delete(node_template3)
+    assert e.value.error.status == 403
+    time.sleep(.5)
+    remove_resource(node_template3)
+
+
+@if_test_rbac
+def test_rbac_node_template_edit(remove_resource):
+    client = get_user_client()
+    node_template = node_template_do(client)
+    # User1 should be able to edit own template
+    name_edit=random_name()
+    client.update(node_template, name=name_edit, 
```

Comment:
Should we pass all of this when doing an update? 
```
digitaloceanConfig=
                  {"region": "nyc3",
                  "size": "2gb",
                   "image": "ubuntu-16-04-x64"}
```

Reply:
We have to pass in the config every time we update a node templated. When trying to update without the config, I was met with an error.